src/django_components/expression.py

```diff
@@ -109,6 +176,37 @@ def is_aggregate_key(key: str) -> bool:
     return ":" in key and not key.startswith(":")
 
 
+# A string that must start and end with quotes, and somewhere inside includes
+# at least one tag. Tag may be variable (`{{ }}`), block (`{% %}`), or comment (`{# #}`).
+DYNAMIC_EXPR_RE = re.compile(
+    r"""^{quote}.*?(?:{var_tag}|{block_tag}|{comment_tag}).*?{quote}$""".format(
+        var_tag=r"(?:\{\{.*?\}\})",
+        block_tag=r"(?:\{%.*?%\})",
+        comment_tag=r"(?:\{#.*?#\})",
+        quote=r"['\"]",
+    )
+)
```

Comment:
This matches strings starting with one quote type, and ending with another: "hallo'

ChatGPT suggests this fix:

```python
DYNAMIC_EXPR_RE = re.compile(
    r"""^(?P<quote>['"]).*?(?:{var_tag}|{block_tag}|{comment_tag}).*?(?P=quote)$""".format(
        var_tag=r"(?:\{\{.*?\}\})",
        block_tag=r"(?:\{%.*?%\})",
        comment_tag=r"(?:\{#.*?#\})",
    )
)

```

Reply:
Very nice! I did check for the quotes mismatch, but separately in the `is_dynamic_expression`. I've updated it to check the quotes inside the regex